src/main/java/io/jenkins/plugins/opentelemetry/job/log/OtelLogStorage.java

```diff
@@ -52,12 +52,9 @@ class OtelLogStorage implements LogStorage {
 
     final OtelTraceService otelTraceService;
 
-    public OtelLogStorage(@NonNull Run run, @NonNull OtelTraceService otelTraceService, @NonNull Tracer tracer) {
+    public OtelLogStorage(@NonNull Run run, @NonNull OtelTraceService otelTraceService, @NonNull Tracer tracer, @NonNull MonitoringAction monitoringAction) {
```

Comment:
if the plugin is not configured (there is no disabled other than un install it), this must not executed, that's the correct solution. The proposed solution hide the issue, if it is really a problem, because you have a plugin installed but not configured.

Reply:
I have to review it if there is another way. The feature of mirroring the logs locally causes more problems than benefits, and for me, it does not make any sense if you configure the sent of logs to an observability backend.

Reply:
I have refactored your changes at https://github.com/jenkinsci/opentelemetry-plugin/pull/881

Reply:
Basically, it is the same solution split in method, and refactoring the return points. Also, I only make changes in the factory; the extra parameter for the OtelLogStorage is unnecessary.